src/hydromt_fiat/workflows/hazard.py

```diff
@@ -1,63 +1,51 @@
 """Hazard workflows."""
 
 import logging
-from pathlib import Path
 
-import geopandas as gpd
 import xarray as xr
-from hydromt import DataCatalog
 from hydromt.model.processes.grid import grid_from_rasterdataset
 
-__all__ = ["hazard_data"]
+__all__ = ["hazard_grid"]
 
 
 logger = logging.getLogger(f"hydromt.{__name__}")
 
 
-def hazard_data(
+def hazard_grid(
     grid_like: xr.Dataset | None,
-    region: gpd.GeoDataFrame,
-    data_catalog: DataCatalog,
-    hazard_fnames: list[str],
-    hazard_type: str | None,
-    return_periods: list[int] | None = None,
+    hazard_data: dict[str : xr.DataArray],
+    hazard_type: str,
     *,
+    return_periods: list[int] | None = None,
     risk: bool,
 ) -> xr.Dataset:
     """Parse hazard data files to xarray dataset.
 
     Parameters
     ----------
-    grid_like: xr.Dataset | None
+    grid_like : xr.Dataset | None
         Grid dataset that serves as an example dataset for transforming the input data
-    region: gpd.GeoDataFrame
-        Region geometry used for reading data from data catalog
-    data_catalog : DataCatalog
-        Model data catalog
-    hazard_fnames : list[str]
-        Names or paths of hazard files
-    hazard_type : str | None
+    hazard_data : dict[str:xr.DataArray]
+        The hazard data in a dictionary with the names of the datasets as keys.
+    hazard_type : str
         Type of hazard
-    risk : bool
-        Designate hazard files for risk analysis
     return_periods : list[int] | None, optional
         List of return periods, by default None
+    risk : bool
+        Designate hazard files for risk analysis
 
     Returns
     -------
     xr.Dataset
         Unified xarray dataset containing the hazard data
     """
     hazard_dataarrays = []
-    for i, hazard_file in enumerate(hazard_fnames):
-        da = data_catalog.get_rasterdataset(hazard_file, geom=region)
-
+    for idx, (da_name, da) in enumerate(hazard_data.items()):
```

Comment:
```suggestion
    for return_period, (da_name, da) in zip(return_periods, hazard_data.items()):
```
Potentially this is personal preference, so I'll leave it up to you whether you prefer this or your original solution, but personally I think this makes it a bit clearer since it's harder to guess what the index will be used for. As far as I can tell it is only used to get the return period on L62, hence the possibility. Either way, up to you

Reply:
Fair enough, will change that